core/src/main/java/org/mule/runtime/core/processor/strategy/SynchronousProcessingStrategyFactory.java

```diff
@@ -45,4 +59,71 @@ public ProcessingStrategy create(MuleContext muleContext, String schedulersNameP
     return SYNCHRONOUS_PROCESSING_STRATEGY_INSTANCE;
   }
 
+  static class SinkPerThreadSink implements Sink, Disposable {
```

Comment:
Maybe, not as clear though.  It's a Sink that uses a Sink per thread.  Let me see i I can come up with another name...